src/db/mod.rs

```diff
@@ -791,9 +791,14 @@ mod database_tests {
     #[cfg(feature = "save_kdbx4")]
     #[test]
     fn test_save() {
-        use crate::db::Entry;
+        use crate::{db::Entry, variant_dictionary::VariantDictionary};
         let mut db = Database::new(Default::default());
 
+        let mut public_custom_data = VariantDictionary::new();
+        public_custom_data.set("example", 42);
```

Comment:
I feel like we're only doing half the test here. Should we not also verify that the value is still there after loading the database? that would make sure not only that the database file is valid after saving with custom data, but that the custom data is still there.

Reply:
There is an [`assert_eq!` check in the test](https://github.com/sseemayer/keepass-rs/blob/c72d53526fde5928505952d3cde6c439f8375622/src/db/mod.rs#L817) to see if the database objects before and after serialization are equal. We `#[derive(PartialEq, Eq)` all the way down to the `VariantDictionary`, so that should be strong enough of a check.

Reply:
:sweat_smile: thanks for pointing that ou